### PR TITLE
feat(exoplayer): add support for IMA SDK

### PR DESCRIPTION
**Pull request by nbennink**
- Added a new prop: `adOptions`. An `adTagUrl` can be added here which is used to request apps.
- Added prop: `onAdEvent` which is called when an ad event is send by IMA.
- Added prop: `onAdError` which is called IMA encountered a fatal error.

### CODE REVIEW

File: Video.js [hunk 1]
```diff
@@ -336,7 +351,7 @@ export default class Video extends Component {
     * Android needs the url to be null sometimes to check if player src has changed. iOS doesn't need this
     * When url is null in iOS the native pip player somehow doesn't work anymore
     */
-    if (!nativeProps.source.uri && Platform.OS === 'ios') return null;
+    if (!nativeProps.source.uri && Platform.OS === 'ios') {return null;}
```
Comment (sjors98) [inline Video.js:354]:
```suggestion
    if (!nativeProps.source.uri && Platform.OS === 'ios') {
        return null;
    }
```

or 
```suggestion
    if (!nativeProps.source.uri && Platform.OS === 'ios') return null;
```

File: android-exoplayer/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java [hunk 2]
```diff
@@ -202,6 +216,14 @@ public ReactExoplayerView(ThemedReactContext context, ReactExoplayerConfig confi
         this.config = config;
         this.bandwidthMeter = config.getBandwidthMeter();
 
+        ImaSdkSettings sdkSettings = ImaSdkFactory.getInstance().createImaSdkSettings();
+        sdkSettings.setAutoPlayAdBreaks(true);
```
Comment (sjors98) [inline android-exoplayer/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java:220]: Isn't this by default? 🤔

File: android-exoplayer/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java [hunk 3]
```diff
@@ -229,6 +251,9 @@ private void createViews() {
         exoPlayerView = new ExoPlayerView(getContext());
         exoPlayerView.setLayoutParams(layoutParams);
 
+        adPlayerView = new PlayerView(getContext());
```
Comment (sjors98) [inline android-exoplayer/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java:254]: Why not the new ExoPlayerView? What is the difference?

File: android-exoplayer/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java [hunk 4]
```diff
@@ -1377,6 +1410,13 @@ public void setDrmLicenseHeader(String[] header){
         exoPlayerView.setIsDrmProtected(header != null && header.length > 0);
     }
 
+    public void setAdOptions(String newAdTagUrl) {
+        if (newAdTagUrl == null) {
```
Comment (sjors98) [inline android-exoplayer/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java:1414]: not sure about java, but will `if(!newAdTagUrl)` work? 
Out of curiousity, I assume you don't want to execute this when newAdTagUrl is undefined or false or something

File: android-exoplayer/src/main/java/com/brentvatne/exoplayer/ReactExoplayerViewManager.java [hunk 5]
```diff
@@ -334,6 +335,12 @@ public void setBufferConfig(final ReactExoplayerView videoView, @Nullable Readab
         }
     }
 
+    @ReactProp(name = PROP_AD_OPTIONS)
+    public void setAdOptions(final ReactExoplayerView videoView, @Nullable ReadableMap adOptions) {
+        String adTagUrl = adOptions.hasKey("adTagUrl") ? adOptions.getString("adTagUrl") : null;
```
Comment (sjors98) [inline android-exoplayer/src/main/java/com/brentvatne/exoplayer/ReactExoplayerViewManager.java:340]: Shouldn't this be `String | null adTagUrl` or something like that?

**Review by sjors98 — APPROVED** [review]

File: android-exoplayer/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java (hunk 2 above)
Comment (nbennink) [reply android-exoplayer/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java:220]: Neen

Comment (sjors98) [reply android-exoplayer/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java:220]: 😢

File: Video.js (hunk 1 above)
Comment (MathijsR94) [reply Video.js:354]: 2nd one please!

**Review by MathijsR94 — APPROVED** [review]

**Pull request merged**